segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexRequestProcessor.java

```diff
@@ -51,10 +60,19 @@ public SearchFailedException(String message) {
      * @param greater boolean to determine if the next higher or the lower value to be returned in case the requested offset is not present.
      *
      * @return the corresponding offset position from the index segment entry.
-     * @throws SearchFailedException if the index segment is of unexpected size or if the search fails.
+     * @throws StreamSegmentNotExistsException If the segment is truncated or it does not exist.
      */
-    public static long locateOffsetForSegment(StreamSegmentStore store, String segment, long targetOffset, boolean greater) throws SearchFailedException {
-        return applySearch(store, segment, targetOffset, greater).getValue();
+    public static long locateOffsetForSegment(StreamSegmentStore store, String segment, long targetOffset, boolean greater) throws StreamSegmentNotExistsException {
+        try {
+            long offset = applySearch(store, segment, targetOffset, greater).getValue();
+            return offset;
+        } catch (SegmentTruncatedException ex) {
+            throw new StreamSegmentNotExistsException(ex.getMessage());
+        } catch (IllegalStateException | SearchFailedException ex) {
+            log.warn("Sending the tail offset for the segment as it failed to locate offset for segment: {}", segment, ex);
+            SegmentProperties segmentProperties = store.getStreamSegmentInfo(segment, TIMEOUT).join();
+            return segmentProperties.getLength();
```

Comment:
This is not the correct value. We should propagate an error.

Reply:
Removed the catch block.